rfcs/0010-define-calculations-of-existential-deposit.md

```diff
@@ -0,0 +1,150 @@
+asset_b_ed_as_native = asset_to_native(asset_b_local_ed, asset_b_ratio)
+native_as_asset_b = native_to_asset(native_asset_ed, asset_b_ratio)
+asset_a_as_asset_b = native_to_asset(asset_a_ed_as_native, asset_b_ratio)
+
+assert(asset_a_ed_as_native == native_asset_ed)
+assert(asset_b_ed_as_native == native_asset_ed)
+assert(asset_a_ed_as_native == asset_b_ed_as_native)
+assert(native_as_asset_b == asset_a_as_asset_b)
+```
+
+### Well-Known Assets & Asset Registry Storage
+
+For a asset to have its ED be correctly calculated, and therefore to exists on 
+chain, it should have its `foreign_asset_ratio` defined in either our Well-Known 
+assets list, or stored in Asset Registry.
+
+## Implementation
+
+### Updates to the Functionality of `multi_existential_deposits`
+
+The current implementation of `multi_existential_deposits` does the following:
+  
+  1. **If an ED is defined for an asset in Asset Registry, return the ED as 
+     is.**
+  
+  2. **Otherwise, pull a hard-coded ED from a `match` statement**.
+  
+  3. If no matches are found, return `Balance::MAX` which will prune unknown
+     assets.
+    
+It should be updated as follows:
+
+  1. **If a asset ratio is defined for an asset in Asset Registry or as a 
+     well-known asset, calculate the standard ED as defined by this document.**
+  
+  3. If no matches are found, return `Balance::MAX` which will prune unknown
+     assets.
+    
+### Existential Deposit Retrieval
+
+For use in the front-end, an RPC will need to be made available that exposes the
+functionality of `multi_existential_deposits`.
+    
+### Asset Registry
+
+**Note**: If AssetRegistry storage is already in use on Picasso, we will need to
+perform a storage migration in the next runtime upgrade for all of these changes
+to be possible.
+
+* The function `AssetRegistry::register_asset` should be updated with the 
+  following signature:
+
+  ```rust
+  pub fn register_asset(
+    origin: OriginFor<T>,
+    location: T::ForeignAssetId,
+    ratio: Ratio, // Used to be optional
+    decimals: Option<Exponent>,
+  ) -> DispatchResultWithPostInfo;
+  ```
+
+## Quality Assurance
+
+* Manually changing the values of different EDs is not supported. By default
+  EDs should behave as defined in this document.
+
+## Questions
+
+* Will inflation/deflation be a concern as assets enter and leave the chain if
+  the ED of a foreign asset is lower/higher on our chain than it is on the
+  assets native chain?
+  
+  > This is not really a solvable problem AFAIK given that we don't control
+    validations done on other chains when transferring to Picasso.
+  
+* Can an RPC expose a function that only exists in the runtime? If not, we may
+  need to move `multi_existential_deposits` (or at least its functionality) into
+  Asset Registry.
+  
+  Moving this functionality into Asset Registry will be necessary for 
+  different runtimes to have different asset configurations.
```

Comment:
Should this be `>`'d?

Reply:
It's not a quote, this was my own summarized of an answer

Reply:
I get where your coming from in terms of formatting Q/A. Will update